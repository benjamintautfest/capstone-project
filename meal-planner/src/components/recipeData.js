import { v4 as uuidv4 } from 'uuid'

const meals = [
    {
        meal: 'Käsestulle',
        id: uuidv4(),
        instructions: 'Stulle schmieren',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
    {
        meal: 'Kartoffelsalat',
        id: uuidv4(),
        instructions: 'Kartoffeln kochen',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
    {
        meal: 'Gemüselasagne',
        id: uuidv4(),
        instructions: 'Lasagne in Ofen',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
    {
        meal: 'Steinpilzrisotto',
        id: uuidv4(),
        instructions: 'Risotto kochen',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
    {
        meal: 'Currywurst',
        id: uuidv4(),
        instructions: 'Wurst braten',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
    {
        meal: 'Linsensuppe',
        id: uuidv4(),
        instructions: 'Dose aufmachen',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
    {
        meal: 'Spaghetti Carbonara',
        id: uuidv4(),
        instructions: 'Nudeln kochen',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
    {
        meal: 'Banana Pancakes',
        id: uuidv4(),
        instructions: 'Pfannekuchen backen',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
    {
        meal: 'Thai Gemüsecurry',
        id: uuidv4(),
        instructions: 'Curry kochen',
        ingredience: ['Käse', 'Stulle', 'Butter'],
    },
]

export default meals
