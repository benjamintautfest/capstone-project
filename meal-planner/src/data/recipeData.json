[
    {
        "id": "01",
        "meal": "Vegetarisches Chili",
        "image": "/images/chili.png",
        "instructions": "Das vegetarische Hack in einer Pfanne mit heißem Öl anbraten. Mit Salz, Pfeffer und Paprikagewürz würzen. Zwiebeln in Ringe schneiden, Paprika stückeln und alles in der Pfanne mit anbraten. Nach Belieben mit Knoblauch und Chili würzen. Tomaten, Tomatenmark und Gemüsebrühe hinzugeben, bei kleiner Hitze ca. 20 Min garen. Bohnen mit Wasser abspülen, mit dem Mais in die Pfanne geben, alles noch etwas weiter köcheln lassen. Mit Salz, Pfeffer, Paprika und Chili kräftig abschmecken.",
        "ingredients": [
            "1 kl. Dose Kidneybohnen",
            "1 Dose geschälte Tomaten",
            "1 kl. Dose Mais",
            "250 g vegetarisches Hack",
            "2 Paprikaschoten",
            "2-3 Zwiebeln"
        ]
    },
    {
        "id": "02",
        "meal": "Kartoffelsalat",
        "image": "/images/chili.png",
        "instructions": "Kartoffeln kochen",
        "ingredients": ["Kartoffeln", "Salat", "Essig"]
    },
    {
        "id": "03",
        "meal": "Gemüselasagne",
        "image": "/images/chili.png",
        "instructions": "Schichten Sie beide Soßen die Nudelplatten und das Gemüse in eine Auflaufform und belegen Sie alles großzügik mit Mozzarella. Überbacken Sie alles, bis es schön schlotzig ist",
        "ingredients": ["Gemüse", "Käse", "Lasagne"]
    },
    {
        "id": "04",
        "meal": "Steinpilzrisotto",
        "image": "/images/chili.png",
        "instructions": "Die getrockneten Steinpilze in eine Schüssel geben und mit heißem Wasser übergießen. Ca. 20 Min. ziehen lassen und anschließend abgießen. Frische Steinpilze in mundgerechte Stüce schneiden. Zwiebel und Knoblauch schälen und fein hacken. Butter in einem großen Topf über mittlerer Hitze schmelzen und Zwiebel und Knoblauch darin ca. 3 Min. anschwitzen.",
        "ingredients": [
            "getrocknete Steinpilze",
            "Zwiebel",
            "Butter",
            "Knoblauch",
            "Risottoreis"
        ]
    },
    {
        "id": "05",
        "meal": "Currywurst",
        "image": "/images/chili.png",
        "instructions": "Wurst braten, mit Paprikapulver und Curry bestreuen, in Ketchup ersäufen und an in tiefem Fett ausgebackenen Kartoffestiften servieren.",
        "ingredients": ["Wurst", "Curry", "Ketchup"]
    },
    {
        "id": "06",
        "meal": "Linsensuppe",
        "image": "/images/chili.png",
        "instructions": "Dose aufmachen",
        "ingredients": ["Linsen", "Suppe", "Löffel"]
    },
    {
        "id": "07",
        "meal": "Spaghetti Carbonara",
        "image": "/images/chili.png",
        "instructions": "Nudeln kochen",
        "ingredients": ["Spaghetti", "Speck", "Eier"]
    },
    {
        "id": "08",
        "meal": "Banana Pancakes",
        "image": "/images/chili.png",
        "instructions": "Eier Mehl und Sosenmilch verquirlen, mit einer Gabel Bananen zerrdrücken und den ganzen Kram in der Pfanne kross aufbraten, Diggi",
        "ingredients": ["Bananen", "Eier", "Mehl"]
    },
    {
        "id": "09",
        "meal": "Thai Gemüsecurry",
        "image": "/images/chili.png",
        "instructions": "Curry kochen",
        "ingredients": ["Gmüse", "Curry", "Reis"]
    }
]
