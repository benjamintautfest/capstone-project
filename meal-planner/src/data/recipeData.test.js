import recipeData from './recipeData.json'

test('delivers the correct recipe data', () => {
    expect(recipeData).toMatchSnapshot()
    expect(recipeData).toHaveLength(9)
    expect(recipeData.map((recipe) => recipe.meal)).toEqual([
        'Gemüselasagne',
        'Vegetarisches Chili',
        'Steinpilzrisotto',
        'Banana Pancakes',
        'Paprika-Zwiebel-Quiche',
        'Käse-Lauch-Suppe',
        'Linsensuppe',
        'Spaghetti Carbonara',
        'Thai Gemüsecurry',
    ])
})

for (let i = 0; i < recipeData.length; i += 1) {
    test(`recipe[${i}] should have properties (id, image, meal, ingredients, instructions)`, () => {
        expect(recipeData[i]).toHaveProperty('id')
        expect(recipeData[i]).toHaveProperty('image')
        expect(recipeData[i]).toHaveProperty('meal')
        expect(recipeData[i]).toHaveProperty('ingredients')
        expect(recipeData[i]).toHaveProperty('instructions')
    })
}

test('mock implementation of a basic function', () => {
    const mock = jest.fn(() => 'i am a mock function')
    expect(mock('calling my mock function !')).toBe('i am a mock function')
    expect(mock).toHaveBeenCalledWith('calling my mock function !')
})

test('mock to return value of a function one time', () => {
    const mock = jest.fn()
    mock.mockReturnValueOnce('Hello')
    mock.mockReturnValueOnce('there')

    mock()
    mock()

    expect(mock).toHaveBeenCalledTimes(2)

    mock('hello', 'there', 'Ben')
    expect(mock).toHaveBeenCalledWith('hello', 'there', 'Ben')

    mock('Ben')
    expect(mock).toHaveBeenLastCalledWith('Ben')
})

test('mock implementation of a function', () => {
    const mock = jest.fn().mockImplementation(() => 'Hamburg')
    expect(mock('Location')).toBe('Hamburg')
    expect(mock).toHaveBeenCalledWith('Location')
})
