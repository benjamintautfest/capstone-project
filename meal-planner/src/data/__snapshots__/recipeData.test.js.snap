// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delivers the correct recipe data 1`] = `
Array [
  Object {
    "id": "01",
    "image": "/images/lasagna.png",
    "ingredients": Array [
      Object {
        "amount": 600,
        "id": "001",
        "item": "Blattspinat",
        "units": "g",
      },
      Object {
        "amount": 2,
        "id": "002",
        "item": "große Karotten",
        "units": "",
      },
      Object {
        "amount": 7,
        "id": "003",
        "item": "große Champignons",
        "units": "",
      },
      Object {
        "amount": 1,
        "id": "004",
        "item": "Zwiebel(n)",
        "units": "",
      },
      Object {
        "amount": 500,
        "id": "005",
        "item": "passierte Tomaten",
        "units": "ml",
      },
      Object {
        "amount": 3,
        "id": "006",
        "item": "Butter",
        "units": "EL",
      },
      Object {
        "amount": 5,
        "id": "007",
        "item": "Milch",
        "units": "EL",
      },
      Object {
        "amount": 1,
        "id": "008",
        "item": "Lasagneplatten",
        "units": "Paket",
      },
    ],
    "instructions": "Karotten, Champignons und Zwiebel schälen, schneiden und zuerst die Zwiebelwürfel in einer Pfanne anbraten, bis sie glasig sind. Danach das restliche Gemüse dazugeben und kurz mitbraten. 5 EL Milch dazugeben und den Rahmspinat darin erhitzen, bis die Blätter nicht mehr gefroren sind und die Rahmsoße leicht eingedickt ist. Jetzt den Frischkäse sowie die Gemüsebrühe dazugeben und gut verrühren. Fünf Minuten köcheln lassen, bis die Soße schön eingedickt ist. Zum Schluss die Knoblauchzehe dazu pressen und mit Salz, Pfeffer und Chili abschmecken. Das Spinatgemisch darf ruhig gut würzig sein, da der Spinat beim Backen relativ viel Salz schluckt",
    "meal": "Gemüselasagne",
  },
  Object {
    "id": "02",
    "image": "/images/chili.png",
    "ingredients": Array [
      Object {
        "amount": 1,
        "id": "009",
        "item": "Kidneybohnen",
        "units": "kl. Dose",
      },
      Object {
        "amount": 1,
        "id": "010",
        "item": "geschälte Tomaten",
        "units": "Dose",
      },
      Object {
        "amount": 1,
        "id": "011",
        "item": "Mais",
        "units": "kl. Dose",
      },
      Object {
        "amount": 250,
        "id": "012",
        "item": "vegetarisches Hack",
        "units": "g",
      },
      Object {
        "amount": 2,
        "id": "013",
        "item": "Paprikaschoten",
        "units": "",
      },
      Object {
        "amount": 3,
        "id": "014",
        "item": "Zwiebel(n)",
        "units": "",
      },
      Object {
        "amount": 5,
        "id": "015",
        "item": "Milch",
        "units": "EL",
      },
      Object {
        "amount": 1,
        "id": "016",
        "item": "Lasagneplatten",
        "units": "Paket",
      },
    ],
    "instructions": "Das vegetarische Hack in einer Pfanne mit heißem Öl anbraten. Mit Salz, Pfeffer und Paprikagewürz würzen. Zwiebeln in Ringe schneiden, Paprika stückeln und alles in der Pfanne mit anbraten. Nach Belieben mit Knoblauch und Chili würzen. Tomaten, Tomatenmark und Gemüsebrühe hinzugeben, bei kleiner Hitze ca. 20 Min garen. Bohnen mit Wasser abspülen, mit dem Mais in die Pfanne geben, alles noch etwas weiter köcheln lassen. Mit Salz, Pfeffer, Paprika und Chili kräftig abschmecken.",
    "meal": "Vegetarisches Chili",
  },
  Object {
    "id": "03",
    "image": "/images/risotto.png",
    "ingredients": Array [
      Object {
        "amount": 1,
        "id": "017",
        "item": "getrocknete Steinpilze",
        "units": "Pck.",
      },
      Object {
        "amount": 1,
        "id": "018",
        "item": "Zwiebel(n)",
        "units": "",
      },
      Object {
        "amount": 3,
        "id": "019",
        "item": "Butter",
        "units": "EL",
      },
      Object {
        "amount": 1,
        "id": "020",
        "item": "Knoblauch",
        "units": "Knolle(n)",
      },
      Object {
        "amount": 1,
        "id": "021",
        "item": "Risottoreis",
        "units": "Pck.",
      },
    ],
    "instructions": "Die getrockneten Steinpilze in eine Schüssel geben und mit heißem Wasser übergießen. Ca. 20 Min. ziehen lassen und anschließend abgießen. Frische Steinpilze in mundgerechte Stüce schneiden. Zwiebel und Knoblauch schälen und fein hacken. Butter in einem großen Topf über mittlerer Hitze schmelzen und Zwiebel und Knoblauch darin ca. 3 Min. anschwitzen.",
    "meal": "Steinpilzrisotto",
  },
  Object {
    "id": "04",
    "image": "/images/pancake.png",
    "ingredients": Array [
      Object {
        "amount": 2,
        "id": "022",
        "item": "Bananen",
        "units": "",
      },
      Object {
        "amount": 2,
        "id": "023",
        "item": "Ei(er)",
        "units": "",
      },
      Object {
        "amount": 110,
        "id": "024",
        "item": "Mehl",
        "units": "g",
      },
      Object {
        "amount": 2,
        "id": "025",
        "item": "Öl",
        "units": "EL",
      },
      Object {
        "amount": 1,
        "id": "026",
        "item": "Salz",
        "units": "Prise",
      },
    ],
    "instructions": "Die Bananen mit der Gabel zerquetschen und die restlichen Zutaten dazugeben, zu einem glatten Teig verrühren. Öl in eine flache (am besten Crêpes-) Pfanne tun und auf dem Herd wie Crêpes ausbacken. Lecker dazu: Nutella oder Marmelade. ",
    "meal": "Banana Pancakes",
  },
  Object {
    "id": "05",
    "image": "/images/quiche.png",
    "ingredients": Array [
      Object {
        "amount": 200,
        "id": "027",
        "item": "Mehl",
        "units": "g",
      },
      Object {
        "amount": 100,
        "id": "028",
        "item": "Butter",
        "units": "g",
      },
      Object {
        "amount": 3,
        "id": "029",
        "item": "Ei(er)",
        "units": "g",
      },
      Object {
        "amount": 1,
        "id": "030",
        "item": "Mineralwasser",
        "units": "Flasche",
      },
      Object {
        "amount": 3,
        "id": "031",
        "item": "Paprikaschoten, rot",
        "units": "",
      },
      Object {
        "amount": 3,
        "id": "032",
        "item": "Zwiebel(n)",
        "units": "",
      },
      Object {
        "amount": 100,
        "id": "033",
        "item": "Crème fraîche",
        "units": "ml",
      },
      Object {
        "amount": 1,
        "id": "034",
        "item": "Tomatenmark",
        "units": "EL",
      },
      Object {
        "amount": 1,
        "id": "035",
        "item": "Käse, gerieben",
        "units": "EL",
      },
    ],
    "instructions": "Aus dem Mehl, der Butter, einem Ei, Mineralwasser und etwas Salz einen Mürbeteig kneten. Danach den Teig in Klarsichtfolie einrollen und im Kühlschrank ca. 1 Stunde ziehen lassen. Jetzt die roten Paprikaschoten entkernen und fein würfeln. Zwiebeln ebenfalls schälen und fein würfeln. Das Gemüse in einer Pfanne bei geschlossenem Deckel so lange anschwitzen, bis es glasig ist. Das dauert ca. 20 Minuten, danach den Herd abschalten und etwas abkühlen lassen. Anschließend den Mürbeteig aus dem Kühlschrank holen, ausrollen und eine gefettete 26-er Springform damit auskleiden. Den Boden mit einer Gabel mehrfach einstechen. Paprika, Zwiebel, 2 Eier, Käse, Ajvar, Tomatenmark und Crème fraîche miteinander vermischen und mit Salz und Pfeffer abschmecken. Die Masse gleichmäßig in der Springform verstreichen. Jetzt die Quiche bei 190 °C Ober-/Unterhitze ca. 40 Minuten backen. Etwas abkühlen lassen vor dem Servieren.",
    "meal": "Paprika-Zwiebel-Quiche",
  },
  Object {
    "id": "06",
    "image": "/images/leeksoup.png",
    "ingredients": Array [
      Object {
        "amount": 300,
        "id": "036",
        "item": "Gemüsebrühe",
        "units": "ml",
      },
      Object {
        "amount": 200,
        "id": "037",
        "item": "Blumenkohl",
        "units": "g",
      },
      Object {
        "amount": 1,
        "id": "038",
        "item": "Lauch",
        "units": "",
      },
      Object {
        "amount": 150,
        "id": "039",
        "item": "Sahneschmelzkäse",
        "units": "g",
      },
      Object {
        "amount": 1,
        "id": "040",
        "item": "Paprikapulver",
        "units": "Pck.",
      },
    ],
    "instructions": "In einem kleinen Topf die Gemüsebrühe zum Kochen bringen und den klein zerstoßenen Blumenkohl etwa 5 min darin kochen. In einem anderen Topf Salzwasser zum Kochen bringen und den Lauch in Ringe geschnitten auch etwa 5 min vorkochen, dann abgießen. In dem Topf mit dem Blumenkohl den Sahnekäse zum Schmelzen bringen, am besten kräftig mit dem Schneebesen einrühren. Suppe würzen. Porree dazugeben, gut verrühren, fertig.",
    "meal": "Käse-Lauch-Suppe",
  },
  Object {
    "id": "07",
    "image": "/images/lentilsoup.png",
    "ingredients": Array [
      Object {
        "amount": 200,
        "id": "041",
        "item": "Linsen",
        "units": "g",
      },
      Object {
        "amount": 4,
        "id": "042",
        "item": "große Kartoffeln",
        "units": "",
      },
      Object {
        "amount": 1,
        "id": "043",
        "item": "Zwiebel(n)",
        "units": "",
      },
      Object {
        "amount": 1,
        "id": "044",
        "item": "Suppengemüse",
        "units": "",
      },
      Object {
        "amount": 1,
        "id": "045",
        "item": "Gemüsesaft",
        "units": "Liter",
      },
      Object {
        "amount": 3,
        "id": "046",
        "item": "Gemüsebrühe",
        "units": "EL",
      },
    ],
    "instructions": "Die Linsen in reichlich Wasser ohne Salz aufkochen und etwa 15 Minuten köcheln lassen. In der Zwischenzeit die Kartoffeln schälen, in kleine Würfel schneiden, das Suppengemüse putzen und in sehr kleine Würfel schneiden, bzw. in einem Multizerkleinerer hacken. Die Linsen in ein Sieb abgießen und abtropfen lassen. In einem großen Topf das Gemüse mit Öl andünsten, Kartoffelwürfel hinzufügen und mit dem Gemüsesaft ablöschen. Die Lorbeerblätter, das Gemüsebrühepulver und die vorgekochten Linsen hinzufügen. Die Linsensuppe solange kochen, bis die Kartoffelwürfel weich sind. Gehackte Petersilie hinzufügen und mit Salz, Pfeffer und wer mag mit reichlich Kümmel abschmecken.",
    "meal": "Linsensuppe",
  },
  Object {
    "id": "08",
    "image": "/images/carbonara.png",
    "ingredients": Array [
      Object {
        "amount": 1,
        "id": "047",
        "item": "Spaghetti",
        "units": "Pck.",
      },
      Object {
        "amount": 200,
        "id": "048",
        "item": "Pancetta",
        "units": "g",
      },
      Object {
        "amount": 4,
        "id": "049",
        "item": "Ei(er",
        "units": "",
      },
      Object {
        "amount": 50,
        "id": "050",
        "item": "Butter",
        "units": "g",
      },
      Object {
        "amount": 1,
        "id": "051",
        "item": "schwarzer Pfeffer",
        "units": "El",
      },
      Object {
        "amount": 300,
        "id": "052",
        "item": "frisch geriebener Parmesan",
        "units": "g",
      },
    ],
    "instructions": "Die Pasta in reichlich Salzwasser bissfest kochen. Den Schinken in Würfel schneiden und in wenig Butter anbraten. Eigelb in einer großen Schüssel mit Salz, Pfeffer und Muskat verquirlen. Die Butter schaumig rühren und gut unter das Eigelb mischen. Die Schinkenwürfel und den geriebenen Käse gründlich unterrühren. Wenn die Nudeln gar sind, abgießen, sofort zu der Mischung in die Schüssel geben, nochmal alles gründlich durchmischen, dann sogleich servieren.",
    "meal": "Spaghetti Carbonara",
  },
  Object {
    "id": "09",
    "image": "/images/curry.png",
    "ingredients": Array [
      Object {
        "amount": "1",
        "id": "053",
        "item": "gemischtes Gemüse",
        "units": " kg",
      },
      Object {
        "amount": 1,
        "id": "054",
        "item": "Zwiebel(n)",
        "units": "",
      },
      Object {
        "amount": 1,
        "id": "055",
        "item": "Knoblauchzehe",
        "units": "g",
      },
      Object {
        "amount": 2,
        "id": "056",
        "item": "Kokosöl",
        "units": "El",
      },
      Object {
        "amount": 500,
        "id": "057",
        "item": "Kokosmilch",
        "units": "ml",
      },
      Object {
        "amount": 500,
        "id": "058",
        "item": "Kokosmilch",
        "units": "ml",
      },
      Object {
        "amount": 2,
        "id": "059",
        "item": "rote Currypaste",
        "units": "El",
      },
      Object {
        "amount": 1,
        "id": "060",
        "item": "Limette",
        "units": "",
      },
      Object {
        "amount": 1,
        "id": "061",
        "item": "Chili",
        "units": "Schote",
      },
    ],
    "instructions": "Gemüse putzen und mundgerecht würfeln. Zwiebel und Knoblauch schälen und fein hacken. Kokosöl in einer großen Pfanne erhitzen. Gemüse circa 10 Minuten braten, dabei regelmäßig wenden. Zwiebel und Knoblauch hinzufügen und weitere 2-3 Minuten garen. Kokosmilch, Currypaste und Limettensaft in die Pfanne geben. 5-10 Minuten einköcheln lassen, dann mit Salz, Pfeffer und Chili abschmecken. Mit Reis servieren.",
    "meal": "Thai Gemüsecurry",
  },
]
`;
